classes/Kohana/Session.php

```diff
@@ -374,8 +374,14 @@ public function write()
 		catch (Exception $e)
 		{
 			// Log & ignore all errors when a write fails
-			Kohana::$log->add(Log::ERROR, Kohana_Exception::text($e))->write();
+			Kohana::$log->error(Kohana_Exception::text($e));
 
+			if (Kohana::$log instanceof Kohana_Logger)
```

Comment:
same comment as above :stuck_out_tongue_winking_eye: to call `Kohana::$log->write();`, we need to be sure that it is a `Kohana_Logger`